ext/opcache/ZendAccelerator.c

```diff
@@ -3800,6 +3800,11 @@ static bool preload_try_resolve_constants(zend_class_entry *ce)
 		ZEND_HASH_MAP_FOREACH_STR_KEY_PTR(&ce->constants_table, key, c) {
 			val = &c->value;
 			if (Z_TYPE_P(val) == IS_CONSTANT_AST) {
+				/* For deprecated constants, we need to flag the zval for recursion
+				 * detection. Make sure the zval is separated out of shm. */
+				if (ZEND_CLASS_CONST_FLAGS(c) & ZEND_ACC_DEPRECATED) {
+					ok = true;
+				}
```

Comment:
I don't quite understand this change. Would this not result in incorrect behavior if there is:

1. A class constant that is `IS_CONSTANT_AST`, but not deprecated and cannot be “updated” (thus setting ok = false).
2. And then another one that is deprecated (thus setting ok = true)?

Reply:
Oh, ofc this is a typo. It was meant to say false. Thanks for catching this.